v1/bank/bank.gunk

```diff
@@ -0,0 +1,355 @@
+//                         },
+//                 },
+//         },
+// }
+package bank
+
+import (
+	"github.com/gunk/opt/http"
+	"github.com/gunk/opt/openapiv2"
+
+	types "github.com/openbank/openbank/v1/types"
+)
+
+// BankRouting holds all the details about a bank routing.
+type BankRouting struct {
+	// Scheme is the routing scheme.
+	Scheme string `pb:"1" json:"scheme"`
+
+	// Address is the routing address.
+	Address string `pb:"2" json:"address"`
+}
+
+// Bank holds all details about a bank.
+type Bank struct {
+	// ID is an identifier for the bank.
+	ID string `pb:"1" json:"id"`
+
+	// FullName is the full name of the bank.
+	FullName string `pb:"2" json:"full_name"`
+
+	// ShortName is the short name of the bank.
+	ShortName string `pb:"3" json:"short_name"`
+
+	// LogoURL is the url for the bank's logo.
+	LogoURL string `pb:"4" json:"logo_url"`
+
+	// WebsiteURL is the url for the bank's website.
+	WebsiteURL string `pb:"5" json:"website_url"`
+
+	// SwiftBIC is the SWIFT bank identifier code.
+	SwiftBIC string `pb:"6" json:"swift_bic"`
+
+	// NationalIdentifier is the national identifier code.
+	NationalIdentifier string `pb:"7" json:"national_identifier"`
+
+	// BankRouting is the routing information for the bank.
+	BankRouting BankRouting `pb:"8" json:"bank_routing"`
+}
+
+// TransactionType is holds the information about transaction type at a bank.
+type TransactionType struct {
+}
+
+// CreateBankRequest is a request envelope to create a bank.
+type CreateBankRequest struct {
+	// Bank is the related information about a bank.
+	Bank Bank `pb:"1" json:"bank"`
+}
+
+// UpdateBankRequest is the request envelope to update a bank.
+type UpdateBankRequest struct {
+	// Bank is the related information about a bank.
+	Bank Bank `pb:"1" json:"bank"`
+}
+
+// DeleteBankRequest is the request envelope to delete a bank.
+type DeleteBankRequest struct {
+	// ID is the bank unique identifier.
+	ID string `pb:"1" json:"id"`
+}
+
+// GetBankRequest is the request envelope to get the bank data.
+type GetBankRequest struct {
+	// ID is the bank unique identifier.
+	ID string `pb:"1" json:"id"`
+
+	// FullName is the full name of the bank.
+	FullName string `pb:"2" json:"full_name"`
+
+	// ShortName is the short name of the bank.
+	ShortName string `pb:"3" json:"short_name"`
+
+	// LogoURL is the url for the bank's logo.
+	LogoURL string `pb:"4" json:"logo_url"`
+
+	// WebsiteURL is the url for the bank's website.
+	WebsiteURL string `pb:"5" json:"website_url"`
+}
+
+// GetBanksResponse is the response for GetBanks
+type GetBanksResponse struct {
+	// Banks is the list of the banks.
+	Banks []Bank `pb:"1" json:"banks"`
+}
+
+// CreateTransactionTypeAtBankRequest is the request envelope to create a new
+// transaction type.
+type CreateTransactionTypeAtBankRequest struct {
+	// TransactionType is the related information about a transaction type at a bank.
+	TransactionType TransactionType `pb:"1" json:"transaction_type"`
```

Comment:
why we have TransactionType as a request and response (L354)

Reply:
Ah, on the request, we are sending this is the body payload, and what is being returned as a response is the same body that we have created through the endpoint.

Reply:
I think it's just going to return what's saved with id.

Reply:
agree with ghazni

Reply:
i think we should have some standard to return create operation, maybe with ID